compiler-rt/lib/sanitizer_common/sanitizer_allocator_local_cache.h

```diff
@@ -166,7 +166,11 @@ struct SizeClassAllocator32LocalCache {
       DCHECK_GT(c->count, 0);
     }
     void *res = c->batch[--c->count];
-    PREFETCH(c->batch[c->count - 1]);
+    // By not doing pointer arithmetic, we avoid the OOB if c->count = 0.
+    // We just prefetch the previous member of the PerClass struct, which
+    // doesn't do harm.
+    PREFETCH(reinterpret_cast<uptr>(c->batch) +
```

Comment:
seems `Max(0, c->count - 1)`  is easier to understand

Reply:
Is the PREFETCH worth it then, if we introduce another branch? Also that doesn't work as is because `c->count` is `uptr`.

Reply:
ah, i guess it's not a real branch on arm64 or x86_64: https://godbolt.org/z/1a11vz6Yr